cpp/arcticdb/version/test/test_version_map_batch.cpp

```diff
@@ -0,0 +1,433 @@
+#include <gtest/gtest.h>
+
+#include <arcticdb/version/version_map_batch_methods.hpp>
+#include <arcticdb/stream/test/stream_test_common.hpp>
+
+using namespace arcticdb;
+using namespace arcticdb::pipelines;
+
+struct VersionMapBatchStore : arcticdb::TestStore {
+protected:
+    std::string get_name() override {
+        return "version_map_batch";
+    }
+};
+
+namespace arcticdb {
+
+AtomKey test_index_key(StreamId id, VersionId version_id) {
+    return atom_key_builder().version_id(version_id).creation_ts(PilotedClock::nanos_since_epoch()).content_hash(3)
+        .start_index(4).end_index(5).build(id, KeyType::TABLE_INDEX);
+}
+
+void add_versions_for_stream(
+    std::shared_ptr<VersionMap> version_map,
+    std::shared_ptr<Store> store,
+    StreamId stream_id,
+    size_t num_versions,
+    size_t start = 0u) {
+    for(auto i = start; i < start + num_versions; ++i) {
+        version_map->write_version(store, test_index_key(stream_id, i));
+    }
+};
+}
+
+TEST_F(VersionMapBatchStore, SimpleVersionIdQueries) {
+    auto store = test_store_->_test_get_store();
+    auto version_map = std::make_shared<VersionMap>();
```

Comment:
These two lines are repeated in every test. We should add them in the fixture as protected members to reduce code duplication.

Reply:
I mind repetition much less in tests than I do in production code. Personally I feel like some of the rules like DRY and no explicit strings that are mandatory for real code are actually reversed in test code, where you want everything to be obvious and local; I don't really want to have to go and look elsewhere to see what a test is doing when it fails.